### PR TITLE
API: Remove interrupt handling and `noprefix.h`

### PR DESCRIPTION
**Pull request by seberg**
We have removed the interrupt handling from the docs a long time ago with a big comment to try to discourage them, so I don't think this should be controversial.
The `noprefix.h` is also so old, that I suspect we can get away with asking users to fix their code, but if anyone thinks otherwise (or maybe rather complains strongly after the fact), I am OK with reconsidering.

Adds a category in the release notes for the removals, since I think the list will get long (likely we should format it as a table at some point, but for now I think I prefer not to have two processes in parallel).

---

**~This is a draft because it is based on top of gh-23916~**

### CODE REVIEW

**Comment by seberg** [comment]
Major downstream packages don't use `noprefix.h` anymore.  Github search finds a few likely abandoned uses, so I guess some folks will see this.  I would tend to try it, but it isn't exactly hard to maintain, so am fine either way.
(interrupt handling is buggy and we don't use it, so I do want to remove it.)

File: pyproject.toml [hunk 1]
```diff
@@ -122,6 +122,11 @@ requires = [
         name = "C API changes"
         showcontent = true
 
+    [[tool.towncrier.type]]
+        directory = "c_api_removal"
+        name = "NumPy 2.0 C API removals"
+        showcontent = true
+
```
Comment (mattip) [inline pyproject.toml:129]: Do you want to add snippets for the removals in #23916?

**Review by mattip — APPROVED** [review]
LGTM. Nice cleanup

**Comment by mattip** [comment]
Thanks @seberg

File: pyproject.toml (hunk 1 above)
Comment (seberg) [reply pyproject.toml:129]: I wasn't going to, because those have all been effectively removed already (use is always an error)

**Pull request merged**